logstash-core/lib/logstash/state_resolver.rb

```diff
@@ -44,10 +44,10 @@ def resolve(pipelines_registry, pipeline_configs)
       configured_pipelines = pipeline_configs.each_with_object(Set.new) { |config, set| set.add(config.pipeline_id.to_sym) }
 
       # If one of the running pipeline is not in the pipeline_configs, we assume that we need to
-      # stop it.
+      # stop it and delete it in registry.
       pipelines_registry.running_pipelines.keys
```

Comment:
We recently had another issue where `running_pipelines` does not include loading pipelines by default, so we added a flag to include those. I think that might be needed here too.

Reply:
good point!